pkg/ebpf/bytecode/runtime/runtime_asset.go

```diff
@@ -40,6 +41,11 @@ type CompiledOutput interface {
 type RuntimeAsset struct {
 	filename string
 	hash     string
+
+	// Telemetry
+	compilerEnabled     int64
```

Comment:
Is this needed here? The fact that we have a `RuntimeAsset` means we have runtime compilation enabled. This should likely be a global flag.

Reply:
The RuntimeAsset is created regardless of whether or not the `config.EnableRuntimeCompiler` flag is enabled: https://github.com/DataDog/datadog-agent/blob/8cf36cd338465c5e333108c3709a6373b6800435/pkg/ebpf/bytecode/runtime/tracer.go#L1-L6

Reply:
We can probably use the fact that `Compile` won't be called if runtime compilation is not enabled. I do think that this flag should be globally reported, and not just part of a `RuntimeAsset`.

Reply:
I'm fine with having it as a global flag, but I do think I need to leave the logic of setting that flag where it is because compilation can fail in `getRuntimeCompiledTracer` before `runtime.Tracer.Compile` is even called, & in that case I think the telemetry should reflect that it was enabled but unsuccessful

Reply:
> compilation can fail in `getRuntimeCompiledTracer` before `runtime.Tracer.Compile` is even called

I think we want to capture that as a "runtime compilation failure". Anytime runtime compilation is turned on, but isn't successful, we want to know. There may be many reasons, but even seeing the numbers in aggregate will be helpful.

Reply:
👍  Sounds good to me, I'll add that field